File: SharedExample/app/src/main/java/com/example/sharedexample/MainActivity.java
package com.example.sharedexample;

import androidx.appcompat.app.AppCompatActivity;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

    private EditText et_save;
    private String shared = "file";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        et_save = (EditText) findViewById(R.id.et_save);

        //앱을 종료했을때도 데이터를 저장(설정 화면에서 많이 쓴다.)
        //앱을 삭제하면 없어진다.
        SharedPreferences sharedPreferences = getSharedPreferences(shared, 0);
        String value = sharedPreferences.getString("home","");//불러오기
        et_save.setText(value);
    }

    @Override
    protected void onDestroy() {//앱(엑티비티)을 종료했을때 실행
        super.onDestroy();

        SharedPreferences sharedPreferences = getSharedPreferences(shared, 0);
        SharedPreferences.Editor editor = sharedPreferences.edit();//sharedPreferences와 연결
        String value = et_save.getText().toString();
        editor.putString("home",value);//저장하기(별명(키), 값)
        editor.commit();//세이브 완료
    }
}

File: testApp/app/src/main/java/com/example/testapp/MainActivity.kt
package com.example.testapp

import android.content.Intent
import android.os.Bundle
import android.preference.PreferenceManager
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        loadData()

        resultButton.setOnClickListener {
            val intent = Intent(this, ResultActivity::class.java)
            val w = weightEditText.text.toString()
            val h = heightEditText.text.toString()
            intent.putExtra("weight",w)
            intent.putExtra("height",h)
            saveData(w.toInt(),h.toInt())
            startActivity(intent)
            //Anko 라이브러리
            //이렇게 해도된다. startActivity<ResultActivity>(
            // "weight" to weightEditText.text.toString(),
            // "height" to heightEditText.text.toString()
            // )

        }
    }

    private fun saveData(w:Int,h:Int ){
        val pref = PreferenceManager.getDefaultSharedPreferences(this)
        val editor = pref.edit()

        editor.putInt("KEY_HEIGHT",h)
            .putInt("KEY_WEIGHT",w)
            .apply()//저장
    }

    private fun loadData(){
        val pref = PreferenceManager.getDefaultSharedPreferences(this)
        val w = pref.getInt("KEY_WEIGHT",0)//저장된 값이 없으면 0을 반환한다.
        val h = pref.getInt("KEY_HEIGHT",0)

        if(w != 0 && h != 0){
            weightEditText.setText(w.toString())
            heightEditText.setText(h.toString())
        }
    }
}
